libs/blocks/icon-block/icon-block.js

```diff
@@ -70,7 +70,15 @@ function decorateContent(el) {
       const textContent = el.querySelectorAll('.text-content > :not(.icon-area)');
       const secondColumn = createTag('div', { class: 'second-column' });
       textContent.forEach((content) => {
-        secondColumn.append(content);
+        let nodeToInsert = content;
+        const firstIcon = content.querySelector('.icon:first-child');
```

Comment:
Just to be sure: you're targeting the icon that is the first sibling within its parent, right? If it had a text before it (or any other element), this would no longer select it and apply the logic. Is that the expectation?